ravenframework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -88,10 +92,16 @@ def run(self, inputIn):
     """
     dataDict = {self.getDataSetName(data): data for _, _, data in inputIn['Data']}
     names = list(dataDict.keys())#
-    evaluation ={k: np.atleast_1d(val) for k, val in self._evaluate(dataDict, **{'dataobjectNames': names}).items()}
+    evaluation ={k: np.atleast_1d(val) for k, val in self._evaluate(inputIn['Data'][0][-1].coords.keys(), dataDict, **{'dataobjectNames': names}).items()}
```

Comment:
> Right now your are using "inputIn['Data'][0][-1].coords.keys()" to determine the type of PCM analysis, either 'static', 'snapshot' or 'Time dependent' analysis. In this case, the PCM post-processor only works for given data that include 'timeTdep' and 'timeSnapshot'. This will significantly limit the flexibility of this PostProcessor. Please define a new XML node in the RAVEN input file to indicate the type of analysis, and use that inside the code to perform corresponding analysis. Please also add checks for each analysis to make the right data is provided from the user.

Docstrings have been added. A new XML node to determine the PCM version has been applied and checks have been added to provide warnings if input data given by users are not in the required format.

Reply:
> Right now your are using "inputIn['Data'][0][-1].coords.keys()" to determine the type of PCM analysis, either 'static', 'snapshot' or 'Time dependent' analysis. In this case, the PCM post-processor only works for given data that include 'timeTdep' and 'timeSnapshot'. This will significantly limit the flexibility of this PostProcessor. Please define a new XML node in the RAVEN input file to indicate the type of analysis, and use that inside the code to perform corresponding analysis. Please also add checks for each analysis to make the right data is provided from the user.

Docstrings have been added. A new XML node to determine the PCM version has been applied and checks have been added to provide warnings if input data given by users are not in the required format.